linchpin/provision/roles/gcloud/tasks/provision_res_defs.yml

```diff
@@ -1,7 +1,7 @@
 ---
 - name: "Set gce_res_type value"
   set_fact:
-    gce_res_type: "{{ res_def['type'] | default(res_def['res_type']) }}"
```

Comment:
now I see the different providers are completely inconsistent.  aws looks for res_type and will fall back to type and this one looks for type and falls back to res_type.

I can see why you want to move to role.   

How did this mess happen to begin with?

Reply:
I explained this a bit above. I hope it's enough.